main/lsp/requests/code_action.cc

```diff
@@ -29,8 +34,234 @@ vector<unique_ptr<TextDocumentEdit>> getEdits(const LSPConfiguration &config, co
+            // We're using root here, because newely defined module would be at the top level
+            return !core::Symbols::root().data(gs)->findMember(gs, nameRef).exists();
+        }
+        return false;
+    };
+
+    // if there are no names like that, return the new module name
+    if (nameNotExists(moduleName)) {
+        return moduleName;
+    }
+
+    // otherwise, try to augment it
+    for (int i = 1; i < 42; i++) {
+        const auto newName = absl::StrCat(moduleName, i);
+        if (nameNotExists(newName)) {
+            return newName;
+        }
+    }
+
+    // bail, if we haven't found untaken name in 42 tries
+    return nullopt;
+}
+
+class MethodCallSiteRenamer : public AbstractRenamer {
+public:
+    MethodCallSiteRenamer(const core::GlobalState &gs, const LSPConfiguration &config, const string oldName,
+                          const string newName)
+        : AbstractRenamer(gs, config, oldName, newName) {
+        const vector<string> invalidNames = {"initialize", "call"};
+        for (auto name : invalidNames) {
+            if (oldName == name) {
+                invalid = true;
+                error = fmt::format("The `{}` method cannot be moved to a module.", oldName);
+                return;
+            }
+        }
+        // block any method not starting with /[a-zA-Z0-9_]+/. This blocks operator overloads.
+        if (!isalnum(oldName[0]) && oldName[0] != '_') {
+            error = fmt::format("The `{}` method cannot be moved to a module.", oldName);
+            invalid = true;
+        }
+    }
+
+    ~MethodCallSiteRenamer() {}
+    void rename(unique_ptr<core::lsp::QueryResponse> &response, const core::SymbolRef originalSymbol) override {
+        if (invalid) {
+            return;
+        }
+        auto loc = response->getLoc();
+
+        // If we're renaming the exact same place twice, silently ignore it. We reach this condition when we find the
+        // same method send through multiple definitions (e.g. in the case of union types)
+        auto it = edits.find(loc);
+        if (it != edits.end()) {
+            return;
+        }
+
+        auto source = loc.source(gs);
+        if (!source.has_value()) {
+            return;
+        }
+        string newsrc;
+        if (auto sendResp = response->isSend()) {
+            // if the call site is not trivial, don't attempt to rename
+            // the typecheck error will guide user how to fix it
+            for (auto dr = sendResp->dispatchResult.get(); dr != nullptr; dr = dr->secondary.get()) {
+                if (dr->main.method != originalSymbol.asMethodRef()) {
+                    return;
+                }
+            }
+
+            edits[sendResp->receiverLoc] = newName;
+        }
+    }
+    void addSymbol(const core::SymbolRef symbol) override {
+        if (symbol.isMethod()) {
+            addSubclassRelatedMethods(gs, symbol.asMethodRef(), getQueue());
+        } else {
```

Comment:
Should we do nothing if the user requests to rename a symbol that is not a method symbol?

Or, if we're sure that we would have early exited already and will never get to this point, can we ENFORCE that `symbol.isMethod()`?

Reply:
This part was cargo culted from the renamer. My understanding was we should add as much symbols as possible, because we use them for the `queryBySymbol` later. I assumed query results may lead to more call site locations
https://github.com/sorbet/sorbet/blob/14bf3533903bd2ea376a323de2657d7b39bf6d20/main/lsp/LSPTask.cc#L324-L342

But I tried adding only method symbols here and all tests passed, so it should be safe to early exit if the symbol is not a method.